cli/src/commands.rs

```diff
@@ -0,0 +1,166 @@
+};
+use clap::{command, Parser, Subcommand};
+use hdp_primitives::{
+    aggregate_fn::{AggregationFunction, FunctionContext},
+    datalake::{
+        block_sampled::BlockSampledCollection,
+        transactions::{IncludedTypes, TransactionsCollection},
+    },
+};
+use std::path::PathBuf;
+
+#[derive(Debug, Parser)]
+#[command(name = "hdp")]
+#[command(version, about, long_about = None)]
+pub struct HDPCli {
+    #[command(subcommand)]
+    pub command: HDPCliCommands,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum HDPCliCommands {
+    /// New to the HDP CLI? Start here!
+    Start,
+    /// Encode the compute and datalake in batch and allow to proceed
+    #[command(arg_required_else_help = true)]
+    Encode {
+        /// Decide to run processor. (default: false)
+        #[arg(short, long, action = clap::ArgAction::SetTrue)]
+        allow_process: bool,
+
+        /// The aggregate function id e.g. "sum", "min", "avg"
+        aggregate_fn_id: AggregationFunction,
+        /// Optional context for applying conditions on the aggregate function "count".
+        /// Format: "{operator}.{value}" (e.g., "eq.100" for equality, "gt.100" for greater-than).
+        /// Supported operators are in the [`Operator`] enum.
+        aggregate_fn_ctx: Option<FunctionContext>,
+
+        #[command(subcommand)]
+        command: DataLakeCommands,
+
+        /// The RPC URL to fetch the datalake
+        rpc_url: Option<Url>,
+
+        /// The chain id to fetch the datalake
+        chain_id: Option<ChainId>,
+
+        /// Path to save output file after pre-process, this is the input file for processor
+        ///
+        /// This will trigger pre-processing step
+        #[arg(short, long)]
+        cairo_input: Option<PathBuf>,
+
+        /// Path to save output file after process
+        ///
+        /// This will trigger processing(=pie generation) step
+        #[arg(short, long, requires("cairo_input"))]
+        output_file: Option<PathBuf>,
+
+        /// Path to save pie file
+        ///
+        /// This will trigger processing(=pie generation) step
+        #[arg(short, long, requires("cairo_input"))]
+        pie_file: Option<PathBuf>,
+    },
+    /// Decode batch computes and datalakes
+    ///
+    /// Note: Batch computes and datalakes should be encoded in bytes[] format
+    #[command(arg_required_else_help = true)]
+    Decode {
+        /// Batched computes bytes
+        #[arg(value_parser = parse_bytes)]
+        tasks: Bytes,
+        /// Batched datalakes bytes
+        #[arg(value_parser = parse_bytes)]
+        datalakes: Bytes,
+    },
+
+    /// Decode one compute and one datalake (not batched format)
+    #[command(arg_required_else_help = true)]
+    DecodeOne {
+        #[arg(value_parser = parse_bytes)]
+        task: Bytes,
+        #[arg(value_parser = parse_bytes)]
+        datalake: Bytes,
+    },
+    /// Run from encoded compute and datalake. Usefull for request batch tasks.
+    Run {
```

Comment:
we need to add long and short versions of flags to make supplying the args more descriptive for sure

Reply:
thats todo for sure, we should do srlsy interface rework, but i'd like to tackle with v2